src/ejabberd_c2s.erl

```diff
@@ -1505,8 +1504,7 @@ terminate(_Reason, StateName, StateData) ->
               <<"Replaced by new connection">>,
               replaced),
             Acc1 = presence_broadcast(Acc0, StateData#state.pres_a, StateData),
-            presence_broadcast(Acc1, StateData#state.pres_i, StateData),
-            reroute_unacked_messages(StateData);
+            presence_broadcast(Acc1, StateData#state.pres_i, StateData);
```

Comment:
why `reroute_unacked_messages/1` is removed?

Reply:
Because this is in the case for session replacement, where the new c2s process is not synchronizing with the old one. `reroute_unacked_messages` is rerouting messages at the same time the new c2s process is already having a stream and delivering messages, therefore having two c2s processes racing each other. I tried explaining that in the description of the PR, that race is what was making tests sometimes fail non-deterministically.
Also, in the replace case, the user didn't request these unacked messages, so why bother with them?

Reply:
how does it correlate with the XEP?